src/main/java/hudson/plugins/performance/PerformanceReportMap.java

```diff
@@ -81,9 +84,15 @@ public void addAll(Collection<PerformanceReport> reports) {
                 }
             }
         }, null);
-        addPreviousBuildReports();
+
+        if (isTopLevel) {
+            loadPreviousBuilds();
```

Comment:
Hi @artem-fedorov 

Would you mind explaining why this `loadPreviousBuilds()` needed to be added?
It wasn't there before, and I can't figure it out from just looking at the code.

Reason I'm asking is that since upgrading the plugin version we're seeing some performance impact with threads spending literally thousands of seconds parsing (see below stack trace fragment from thread dump), presumably going through the entire job history.

```
[...]
hudson.plugins.performance.parsers.AbstractParser.initDateFormat(AbstractParser.java:225)
hudson.plugins.performance.parsers.AbstractParser.parseTimestamp(AbstractParser.java:207)
hudson.plugins.performance.parsers.JMeterCsvParser.getSample(JMeterCsvParser.java:136)
hudson.plugins.performance.parsers.JMeterCsvParser.parseCSV(JMeterCsvParser.java:80)
hudson.plugins.performance.parsers.JMeterCsvParser.parse(JMeterCsvParser.java:70)
hudson.plugins.performance.parsers.AbstractParser.parse(AbstractParser.java:85)
hudson.plugins.performance.PerformanceReportMap.parseReports(PerformanceReportMap.java:491)
hudson.plugins.performance.PerformanceReportMap.<init>(PerformanceReportMap.java:78)
hudson.plugins.performance.actions.PerformanceBuildAction.getPerformanceReportMap(PerformanceBuildAction.java:87)
hudson.plugins.performance.PerformanceReportMap.getReportMap(PerformanceReportMap.java:534)
hudson.plugins.performance.PerformanceReportMap.loadPreviousBuilds(PerformanceReportMap.java:506)
hudson.plugins.performance.PerformanceReportMap.<init>(PerformanceReportMap.java:89)
hudson.plugins.performance.actions.PerformanceBuildAction.getPerformanceReportMap(PerformanceBuildAction.java:87)
hudson.plugins.performance.actions.PerformanceBuildAction.getTarget(PerformanceBuildAction.java:62)
hudson.plugins.performance.actions.PerformanceBuildAction.getTarget(PerformanceBuildAction.java:18)
[...]
```

Reply:
@tilln 
Hi. I can't be sure now, but we need load all reports for create Trend history

Reply:
Well, it seems to work with that line removed though.

Reply:
did you test it after restart Jenkins server or open other job?

Reply:
Definitely after restart (needed for plugin redeployment).

Reply:
Worth noting is also that serialized report files seem to get parsed over and over again, i.e. thread dumps show `JMeterCsvParser.parse` instead of `AbstractParser.loadSerializedReport`. No idea what would have caused that to change.

Reply:
@artem-fedorov 
I have done some more digging and the performance impact seems to be caused by 2 compounding things (in conjunction with a significant build history and not small CSV files):
1. A bug in `AbstractParser.loadSerializedReport` that makes the report caching ineffective and leads to parsing for previously cached reports.
2. The "new" `loadPreviousBuilds()` loads more than previously, frequently hitting the cache (leading to reparsing). 
Also, it appears to be redundant as `addPreviousBuildReports()` loads what is needed.

I'll have submitted PRs #188 and #189 so they can be reviewed individually.